perun-core/src/test/java/cz/metacentrum/perun/core/entry/MembersManagerEntryIntegrationTest.java

```diff
@@ -1714,6 +1717,32 @@ public void findOnlySponsoredMembers() throws Exception {
 		assertTrue(members.size() == 0);
 	}
 
+	@Test
+	public void removeLastSponsorWithoutExpiration() throws Exception {
+		System.out.println(CLASS_NAME + "removeLastSponsorWithoutExpiration");
+
+		//Set up sponsor
+		Member sponsorMember = setUpSponsor(createdVo);
+		User sponsorUser = perun.getUsersManagerBl().getUserByMember(sess, sponsorMember);
+		AuthzResolverBlImpl.setRole(sess, sponsorUser, createdVo, Role.SPONSOR);
+		//Set up expiration rule
+		Map<String, String> rulesMap = new LinkedHashMap<>();
+		rulesMap.put("doNotExpireSponsoredMembers", "true");
+		Attribute attribute = perun.getAttributesManagerBl().getAttribute(sess, createdVo, expirationRulesAttr);
+		attribute.setValue(rulesMap);
+		perun.getAttributesManagerBl().setAttribute(sess, createdVo, attribute);
+		//create sponsored member
+		Map<String, String> nameOfUser1 = new HashMap<>();
+		nameOfUser1.put("guestName", "Ing. Jiří Novák, CSc.");
+		Member sponsoredMember = perun.getMembersManagerBl().createSponsoredMember(sess, createdVo, "dummy", nameOfUser1, "secret", sponsorUser, false);
+		//Remove sponsor
+		perun.getMembersManagerBl().removeSponsor(sess, sponsoredMember, sponsorUser);
+		//refresh from DB
+		sponsoredMember = perun.getMembersManagerBl().getMemberById(sess, sponsoredMember.getId());
+
+		assertNotSame("Sponsored member without sponsor cannot expire when doNotExpireSponsoredMembers rule is set to true", sponsoredMember.getStatus(), Status.EXPIRED);
+	}
```

Comment:
Please add also test which checks this behavior without attribute set (or with the false in value set) if not exists yet.

Reply:
There is already a test, which checks the situation when the attribute is not set.